config/config-builder/src/swarm_config.rs

```diff
@@ -2,221 +2,112 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! Convenience structs and functions for generating configuration for a swarm of libra nodes
-use failure::prelude::*;
+use failure::{bail, ensure, Result};
 use libra_config::{
-    config::{
-        NetworkConfig, NodeConfig, OnDiskStorageConfig, PersistableConfig, RoleType,
-        SafetyRulesBackend, SeedPeersConfig, VMPublishingOption,
-    },
-    trusted_peers::{
-        ConsensusPeerInfo, ConsensusPeersConfig, NetworkPeerInfo, NetworkPeersConfig,
-        UpstreamPeersConfig,
-    },
-    utils,
+    config::{NodeConfig, PersistableConfig, RoleType},
+    generator,
 };
 use libra_crypto::{
     ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
     test_utils::KeyPair,
 };
 use libra_types::transaction::Transaction;
-use rand::{rngs::StdRng, SeedableRng};
-use std::{
-    collections::HashMap,
-    path::{Path, PathBuf},
-};
+use std::path::{Path, PathBuf};
 use vm_genesis;
 
-pub struct SwarmConfig {
```

Comment:
What's the rationale for getting ride of the SwarmConfig type? I think it makes more sense to have the type unless i'm missing something

Reply:
The type contains a single resource. It seems cleaner to just pass that resource instead of bundling it inside another resource that adds a layer of ambiguity. Alternatively we could use an alias.

Reply:
i disagree about it adding a layer of ambiguity and I don't believe an alias is the right thing. There's also great value with having a thin wrapper around another type, its a paradigm that is used a lot in rust and other functional and adds a great deal of value

Reply:
I brought this back in a commit on top so as to not enter rebase hell.